crates/sargon/src/profile/mfa/secured_entity_control/secured_entity_control.rs

```diff
@@ -26,13 +26,103 @@ pub struct SecuredEntityControl {
     /// The believed-to-be-current security structure of FactorInstances which
     /// secures this entity.
     pub security_structure: SecurityStructureOfFactorInstances,
+
+    /// A provisional new security structure configuration which user
+    /// is about to change to
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub provisional: Option<ProvisionalSecurifiedConfig>,
+}
+
+impl HasProvisionalSecurifiedConfig for SecuredEntityControl {
+    fn get_provisional(&self) -> Option<ProvisionalSecurifiedConfig> {
+        self.provisional.clone()
+    }
+
+    fn set_provisional_unchecked(
+        &mut self,
+        provisional: impl Into<Option<ProvisionalSecurifiedConfig>>,
+    ) {
+        self.provisional = provisional.into();
+    }
+}
+
+pub trait HasProvisionalSecurifiedConfig {
+    fn get_provisional(&self) -> Option<ProvisionalSecurifiedConfig>;
+
+    /// # Throws
+    /// Throws if the provisional is already `Some(ProvisionalSecurifiedConfig::TransactionQueued)`,
+    /// you need to first cancel the transaction and then call the method `cancel_queued_transaction` on
+    /// this type
+    fn set_provisional(
+        &mut self,
+        provisional: impl Into<Option<ProvisionalSecurifiedConfig>>,
+    ) -> Result<()> {
+        let provisional = provisional.into();
+        let maybe_existing = self.get_provisional();
+        let Some(existing) = maybe_existing.as_ref() else {
+            return self.set_provisional_unchecked_ok(provisional);
+        };
+        match existing {
+            ProvisionalSecurifiedConfig::FactorInstancesDerived { value: _ } => {
+                if let Some(new) = provisional.as_ref() {
+                    if new.is_transaction_queued() {
+                        // We allow `FactorInstancesDerived` -> `TransactionQueued` transition ofc...
+                        return self.set_provisional_unchecked_ok(provisional);
+                    }
+                }
+                // We have already consumed FactorInstances from the FactorInstancesCache.
+                // We currently do not have a way of putting these back. We don't want to 
+                // create gaps if we can. Wallet Features might change, to REQUIRE us to
+                // allow this, if so we can change this - allowing transition from 
+                // `FactorInstancesDerived` to Non-TransactionQueued state.
+                Err(
+                CommonError::SecurityEntityControlCannotChangeProvisionalAlreadyDerivedInstances,
```

Comment:
This can happen quite easily if user updates a shield a second time before applying it. So most likely we will need to handle this, let's discuss in design session how to do it.